### PR TITLE
Add addJqueryUI and addJqueryPlugin methods in LegacyControllerContext

### LINKED ISSUE #36649: Methods addJqueryPlugin / addJqueryUI are not available on migrated controllers when used in displayBackOfficeHeader/actionAdminControllerSetMedia hooks
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hello,

I'm facing an exception when attempting to use the `addJqueryPlugin`/`addJqueryUI` methods in the `displayBackOfficeHeader`/`actionAdminControllerSetMedia` hooks on the new product page:

```Error rendering "HeadTag" component: Call to undefined method ContainerXXXXX\LegacyControllerContextProxyXXXXXX::addJqueryUI()```

It appears the methods are not defined in the `LegacyControllerContext` (src/Core/Context/LegacyControllerContext.php) unlike addJS/addCSS which are available and can be used without issue.

Copying the methods from the `Controller` class to the `LegacyControllerContext` class appears to solve the issue and the plugins are properly added.

I believe this should be kept in v9 as these methods are still used by module developers to import jQuery UI/plugins present in the root `/js` folder of PrestaShop, which remains more convenient than having modules ship their own scripts.

### Expected behavior

These method should remain available in order to allow module developers to include these native assets on the migrated controller pages.

### Steps to reproduce

- Register a module on either `actionAdminControllerSetMedia` or `displayBackOfficeHeader`
- Have said method call `$this->context->controller->addJqueryUI(['ui.slider'])` or `$this->context->controller->addJqueryPlugin('tablednd')`
- Enter a migrated controller like the new product edition page, and see the exception:
```Error rendering "HeadTag" component: Call to undefined method ContainerXXXX\LegacyControllerContextProxyXXXXX::addJqueryUI()```

### PrestaShop version(s) where the bug happened

develop

### PHP version(s) where the bug happened

8.1

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

PrestaModule

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add addJqueryUI and addJqueryPlugin methods in LegacyControllerContext, plus a small UX bug fixed in module configuration page
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI green and UI tests green<br>See how to test section
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/10773040026
| Fixed issue or discussion?     | Fixes #36649
| Related PRs       | ~
| Sponsor company   | ~

### How to test

Edit the `hookActionAdminControllerSetMedia` method in the `modules/blockwishlist/blockwishlist.php` module Add this code:

```php
        $this->context->controller->addJqueryUi('ui.widget');
        $this->context->controller->addJqueryPlugin('tagify');
```

Now go to any migrated page you should have an error referring to not knowing the `addJqueryUi`. With this PR the error no longer exists.

### CODE REVIEW

**Review by kpodemski — DISMISSED** [review]

**Review by boherm — APPROVED** [review]

File: phpstan-baseline.neon
Comment (kpodemski) [inline phpstan-baseline.neon:220]: 😎

**Review by kpodemski — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @jolelievre ,

Tested your PR following your steps to reproduce. Tested with and without debug mode.

Before I have an exception :
Screenshot 2024-09-09 at 16 49 47

With the PR, I have access to any migrated BO pages.

Thanks!
It is QA ✅

**Comment by ps-jarvis** [comment]
PR merged, well done!

Message to @PrestaShop/committers: do not forget to milestone it before the merge.

**Pull request merged**
